### DISCUSSION TITLE
[prob 2.3]definition about recall@10

### DISCUSSION THREAD

**Discussion by bb0711** [seominjoon/kaist-ai605#126] [Q&A] [answered]
1.  I'm confused about the definition of recall.
Does it right that recall@10 = ( # of ground truth document among 10 predicted documents) / ( # of ground truth document) ?
I think almost all the questions have 1 ground truth document, so recall@10 for each query is 0 or 1 right?
Or recall@10 = tp/(tp+tn) which is always smaller than 1/10 (after removing duplicated documents) ?

2. For sparse search, does it okay to use numpy for computing similarity or tfidf things?

3. For density search, when averaging word embedding, do we have to ignore UNKNOWN words or also consider them? 

Thank you in advance :)

**Comment by seominjoon** [comment, marked as answer]
1. you are right that recall at 10 is 0 or 1 for each document. You will need to average it over many queries.
2. It is okay in that you are allowed to use it, but it probably won't work (using too much memory).
3. There won't be word embedding corresponding to UNK so you should just ignore it.
